GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -80,15 +83,43 @@ public override void Stop()
             string sha)
         {
             // TODO(Mac): Add functional tests that validate file mode is set correctly
-            ushort fileMode = this.FileSystemCallbacks.GitIndexProjection.GetFilePathMode(relativePath);
-            Result result = this.virtualizationInstance.WritePlaceholderFile(
-                relativePath,
-                PlaceholderVersionId,
-                ToVersionIdByteArray(FileSystemVirtualizer.ConvertShaToContentId(sha)),
-                (ulong)endOfFile,
-                fileMode);
+            FileTypeAndMode fileTypeAndMode = this.FileSystemCallbacks.GitIndexProjection.GetFileTypeAndMode(relativePath);

```

Comment:
As far as API goes, is there any reason FileType and Mode should be tied together? Or are we still leaking private details of the index format?

I think this `FileTypeAndMode` class is good for storing the values internally, but seeing this usage, I'm wondering if we should just have separate `GetFileType` and `GetMode` methods. What do you think?

Overall, I'd say this is pretty minor now, since at least the calling code isn't dealing with masks.